tasks/main.yml

```diff
@@ -26,75 +26,18 @@
         (ansible_python_version is version('3.12', '>=')) and
         (docker_apt_key_url is match('https://'))
 
-- name: Install Docker APT deps
-  ansible.builtin.package:
-    name: "{{ packages }}"
-    state: present
-  vars:
-    packages:
-      - apt-transport-https
-      - ca-certificates
-      - gnupg2
-      - software-properties-common
-
-# Link: https://docs.docker.com/engine/install/debian/#install-using-the-repository
-- name: Prepare apt with key and sources entry
-  block:
-    - name: Ensure apt keyrings dir is present
-      ansible.builtin.file:
-        path: "{{ docker_apt_keyrings_dir }}"
-        state: directory
-        owner: root
-        group: root
-        mode: '0755'
-
-    - name: Put docker apt release key in place
-      ansible.builtin.get_url:
-        url: "{{ docker_apt_key_url }}"
-        dest: "{{ docker_apt_keyrings_dir }}/docker.asc"
-        mode: '0644'
-        backup: true
-        force: true
-      register: apt_repo
-
-    - name: Add Docker's APT repository
-      ansible.builtin.template:
-        src: templates/docker.list.j2
-        dest: /etc/apt/sources.list.d/docker.list
-        owner: root
-        group: root
-        mode: '0644'
-      register: apt_repo
-
-    # that key was created in 2017 and was still in use in 2024,
-    # we most probably need to remove this one key only
-    - name: Remove key from deprecated apt-key storage
-      ansible.builtin.apt_key:
-        id: 0x9DC858229FC7DD38854AE2D88D81803C0EBFCD88
-        state: absent
-
-- name: Optionally get and check GPG fingerprint of apt key file
-  when: docker_apt_key_fpr is defined   # no need to get the fpr if user does not want to check it anyways
-  block:
-    - name: Get GPG fingerprint of downloaded key file
-      ansible.builtin.shell:
-        cmd: "gpg --with-colons --show-keys '{{ docker_apt_keyrings_dir }}/docker.asc' | grep -A1 '^pub:' | grep '^fpr:' | cut -d: -f10"
-      changed_when: false
-      register: gpg_fpr
+- name: Determine Trixie-style APT setup
+  ansible.builtin.set_fact:
+    # Debian Trixie has release number 13.
+    trixie_setup: "{{ ansible_distribution == 'Debian' and ansible_distribution_version | int >= 13 }}"
```

Comment:
I suggest testing against `ansible_distribution_major_version` here or use [ansible.builtin.version test – compare version strings](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/version_test.html), because `ansible_distribution_version` usually contains something like "12.11" (major.minor).  See the following excerpt from `ansible_facts`:

```json
    "ansible_distribution": "Debian",
    "ansible_distribution_file_parsed": true,
    "ansible_distribution_file_path": "/etc/os-release",
    "ansible_distribution_file_variety": "Debian",
    "ansible_distribution_major_version": "12",
    "ansible_distribution_minor_version": "11",
    "ansible_distribution_release": "bookworm",
    "ansible_distribution_version": "12.11",
```